libs/langchain/langchain/schema/runnable/configurable.py

```diff
@@ -271,10 +280,14 @@ class RunnableConfigurableAlternatives(DynamicRunnable[Input, Output]):
 
     @property
     def config_specs(self) -> Sequence[ConfigurableFieldSpec]:
-        which_enum = StrEnum(  # type: ignore[call-overload]
-            self.which.name or self.which.id,
-            ((v, v) for v in list(self.alternatives.keys()) + [self.default_key]),
-        )
+        if which_enum := _enums_for_spec.get(self.which):
```

Comment:
Do we need locks in case code is used from a threaded context?

Reply:
that's the reason why I made this a single atomic operation, instead of 
```
if dict.get(key):
  value = dict[key]
```

Reply:
The switching is handled by the OS, does this actually map to an atomic operation as far as the OS is concerned?